File: src/telegram/types/poll.ts
import * as type from '../types'
import Type from './type'




export declare type JsonPoll = {
	id: type.String
	question: type.String
	options: type.JsonPollOption[]
	total_voter_count: type.Number
	is_closed: type.Boolean
	is_anonymous: type.Boolean
	type: type.PollType
	allows_multiple_answers: type.Boolean
	correct_option_id?: type.Number
	explanation?: type.String
	explanation_entities?: type.JsonMessageEntity[]
	open_period?: type.Number
	close_date?: type.Number
}


export default class Poll extends Type {
	id: type.String
	question: type.String
	options: type.PollOption[]
	total_voter_count: type.Number
	is_closed: type.Boolean
	is_anonymous: type.Boolean
	type: type.PollType
	allows_multiple_answers: type.Boolean
	correct_option_id?: type.Number
	explanation?: type.String
	explanation_entities?: type.MessageEntity[]
	open_period?: type.Number
	close_date?: type.Number

	defaultObject!: JsonPoll
	constructor(options: JsonPoll) {
		super()

		this.defaultObject = options

		this.id = options.id
		this.question = options.question
		this.options = options.options.map(o => new type.PollOption(o))
		this.total_voter_count = options.total_voter_count
		this.is_closed = options.is_closed
		this.is_anonymous = options.is_anonymous
		this.type = options.type
		this.allows_multiple_answers = options.allows_multiple_answers
		this.correct_option_id = options.correct_option_id || undefined
		this.explanation = options.explanation || undefined
		this.explanation_entities = options.explanation_entities && options.explanation_entities.map(e => new type.MessageEntity(e)) || undefined
		this.open_period = options.open_period || undefined
		this.close_date = options.close_date || undefined
	}
}

File: telegram-api-json/pygen/Poll.py

from telelib.bot import DefaultType

# @url https://core.telegram.org/bots/api#poll# ['This object contains information about a poll.']
class Poll(DefaultType):

    
	# @var id Unique poll identifier
	@property
	def id(self) -> String:
		return self._d["id"]
	# @var question Poll question, 1-300 characters
	@property
	def question(self) -> String:
		return self._d["question"]
	# @var options List of poll options
	@property
	def options(self) -> Array of PollOption:
		return self._d["options"]
	# @var total_voter_count Total number of users that voted in the poll
	@property
	def total_voter_count(self) -> Integer:
		return self._d["total_voter_count"]
	# @var is_closed True, if the poll is closed
	@property
	def is_closed(self) -> Boolean:
		return self._d["is_closed"]
	# @var is_anonymous True, if the poll is anonymous
	@property
	def is_anonymous(self) -> Boolean:
		return self._d["is_anonymous"]
	# @var type Poll type, currently can be "regular" or "quiz"
	@property
	def type(self) -> String:
		return self._d["type"]
	# @var allows_multiple_answers True, if the poll allows multiple answers
	@property
	def allows_multiple_answers(self) -> Boolean:
		return self._d["allows_multiple_answers"]
	# @var correct_option_id Optional. 0-based identifier of the correct answer option. Available only for polls in the quiz mode, which are closed, or was sent (not forwarded) by the bot or to the private chat with the bot.
	@property
	def correct_option_id(self) -> Integer:
		return self._d["correct_option_id"]
	# @var explanation Optional. Text that is shown when a user chooses an incorrect answer or taps on the lamp icon in a quiz-style poll, 0-200 characters
	@property
	def explanation(self) -> String:
		return self._d["explanation"]
	# @var explanation_entities Optional. Special entities like usernames, URLs, bot commands, etc. that appear in the explanation
	@property
	def explanation_entities(self) -> Array of MessageEntity:
		return self._d["explanation_entities"]
	# @var open_period Optional. Amount of time in seconds the poll will be active after creation
	@property
	def open_period(self) -> Integer:
		return self._d["open_period"]
	# @var close_date Optional. Point in time (Unix timestamp) when the poll will be automatically closed
	@property
	def close_date(self) -> Integer:
		return self._d["close_date"]

